### PR TITLE
feat: reconcile syncset for enterprise clusters

### PR DESCRIPTION
**Pull request by pb82**
## Description

Reconcile the SyncSet for Enterprise clusters using ocm. Compared to managed clusters, the SyncSet will only contain the Observability and Pull Secret pieces.

## Verification Steps

1. Register an enterprise cluster
2. Make sure that it reaches 'ready' status and that only the SyncSet is reconciled.
3. Make sure that the Observability Operator is running on the cluster.
4. Step 3 also ensures that the Pull Secrets have been installed.

### CODE REVIEW

**Review by machi1990 — APPROVED** [review]
the code change lgtm

I do not have a cluster handy to test. @pb82 & @pawelpaszki have you been able to verify it against a live cluster?

**Comment by pb82** [comment]
@machi1990 yes, we have verified that against a cluster. Pawel suggested to verify again against a cluster that is not ready to see how KFM would behave.

**Comment by machi1990** [comment]
> @machi1990 yes, we have verified that against a cluster. Pawel suggested to verify again against a cluster that is not ready to see how KFM would behave.

+1 great suggestion @pawelpaszki

File: internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go [hunk 1]
```diff
@@ -725,17 +744,30 @@ func (c *ClusterManager) reconcileClusterWithManualConfig() []error {
...
+	switch cluster.ClusterType {
+	case api.EnterpriseDataPlaneClusterType.String():
+		r = append(r, []interface{}{
+			c.buildObservabilityNamespaceResource(),
+			c.buildObservatoriumDexSecretResource(),
+			c.buildObservatoriumSSOSecretResource(),
+			c.buildObservabilityCatalogSourceResource(),
+			c.buildObservabilityOperatorGroupResource(),
+			c.buildObservabilitySubscriptionResource(),
+		}...)
```
Comment (ziccardi) [inline internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:757]:
```suggestion
		r = append(r, 
			c.buildObservabilityNamespaceResource(),
			c.buildObservatoriumDexSecretResource(),
			c.buildObservatoriumSSOSecretResource(),
			c.buildObservabilityCatalogSourceResource(),
			c.buildObservabilityOperatorGroupResource(),
			c.buildObservabilitySubscriptionResource(),
		)
```

File: internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go [hunk 2]
```diff
@@ -725,17 +744,30 @@ func (c *ClusterManager) reconcileClusterWithManualConfig() []error {
...
+			c.buildDedicatedReaderClusterRoleBindingResource(),
+			c.buildKafkaSREGroupResource(),
+			c.buildKafkaSreClusterRoleBindingResource(),
+			c.buildObservabilityNamespaceResource(),
+			c.buildObservatoriumDexSecretResource(),
+			c.buildObservatoriumSSOSecretResource(),
+			c.buildObservabilityCatalogSourceResource(),
+			c.buildObservabilityOperatorGroupResource(),
+			c.buildObservabilitySubscriptionResource(),
+		}...)
```
Comment (ziccardi) [inline internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go:770]:
```suggestion
		r = append(r, 
			c.buildReadOnlyGroupResource(),
			c.buildDedicatedReaderClusterRoleBindingResource(),
			c.buildKafkaSREGroupResource(),
			c.buildKafkaSreClusterRoleBindingResource(),
			c.buildObservabilityNamespaceResource(),
			c.buildObservatoriumDexSecretResource(),
			c.buildObservatoriumSSOSecretResource(),
			c.buildObservabilityCatalogSourceResource(),
			c.buildObservabilityOperatorGroupResource(),
			c.buildObservabilitySubscriptionResource(),
		)
```

**Comment by machi1990** [comment]
I'll verify this once I've a cluster up and running @pb82 @pawelpaszki

**Comment by machi1990** [comment]
> I'll verify this once I've a cluster up and running @pb82 @pawelpaszki

I am from executing the test plan as explained in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1454#issuecomment-1380684827 and everything works as expected. @pb82 @pawelpaszki 

@pb82 do you mind addressing @ziccardi comments and then we'll be able to merge this? Thanks

**Review by ziccardi — APPROVED** [review]
lgtm

**Pull request merged**